File: day2/C/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <list>

using namespace std;

int Data[117];

void readFile(string FileName){ //Read in the data file
    ifstream MyFile(FileName);
    string Line;
    int Temp;
    int i = 0;
    if (MyFile.is_open()){
        cout << "file openend" << endl;
        while (! MyFile.eof()){
            Line = "";
            Temp = 0;
            getline(MyFile, Line); //read line
            istringstream TempInt (Line); //convert string to int
            TempInt >> Temp;
            Data[i] = Temp; //save data
            i++;
        }
        MyFile.close();
    }
    else {cout << "No file found";}
}



void addOP(int Index){
    int Pos1 = Data[Index+1];
    int Pos2 = Data[Index+2];
    int store = Data[Index+3];

    Data[store] = Data[Pos1] + Data[Pos2];   
}

void multOP(int Index){
    int Pos1 = Data[Index+1];
    int Pos2 = Data[Index+2];
    int store = Data[Index+3];

    Data[store] = Data[Pos1] * Data[Pos2];   
}

int main(){
    readFile("input.txt");
    for (int i = 0; i < 117; i = i + 4){
        cout << Data[i] << " ";
        if (Data[i] == 99){
            cout << "Done" << endl;
            for (int x=0; x < 117; ++x){
                cout << Data[x] << " ";
            }
            cout << endl << Data[0];
            return 0;
        }else if (Data[i] == 1){
            addOP(i);
        }else if(Data[i] == 2){
            multOP(i);
        }
    }
    return 0;
}

File: day2/Python/main.py

def readFile():
    F = open("input.txt", "r")
    Data = []
    for Line in F:
        Data.append(int(Line.rstrip()))
    return Data

def addOP(Index, Data):
    Pos1 = Data[Index + 1]
    Pos2 = Data[Index + 2]
    Store = Data[Index + 3]

    Data[Store] = Data[Pos1] + Data[Pos2]
    return Data

def multOP(Index, Data):
    Pos1 = Data[Index + 1]
    Pos2 = Data[Index + 2]
    Store = Data[Index + 3]

    Data[Store] = Data[Pos1] * Data[Pos2]
    return Data

def main():
    for x in range(0,99):
        for y in range(0,99):
            Data = readFile()
            Data[1] = x
            Data[2] = y
            for i in range(0, len(Data), 4):
                if Data[i] == 99:
                    break
                if Data[i] == 1:
                    Data = addOP(i, Data)
                elif Data[i] == 2:
                    Data = multOP(i, Data)
            if Data[0] == 19690720:
                print(f"Noun {Data[1]}, Verb {Data[2]}")
                print(Data[1]*100+Data[2])

    print(f"Data 0: {Data[3]}")
    print(i)
    return None

if __name__ == "__main__":
    main()
